### ISSUE TITLE
non-quoted paths can cause output data retrieval from_work_dir to fail

### ISSUE THREAD

**Issue by wm75** [galaxyproject/galaxy#8711]
During tool development I found that, currently, job execution chokes on elements like
```
<data name="coverage_profile_along_highly_expressed_genes" format="tsv" from_work_dir="results/coverage_profile_along_genes_(high).txt" />
```
in a wrapper's `<outputs>` section.

The reason is the parentheses in the name of the output file generated by the underlying tool, which combined with failure to quote paths in `command_factory.__copy_if_exists_command`:
https://github.com/galaxyproject/galaxy/blob/7d1beeb1a29b35de3b4fff384efa23e85d74f07c/lib/galaxy/jobs/command_factory.py#L235
causes a `line 60: syntax error near unexpected token '('` in the job execution script.

Question is: is this worth fixing? It only affects a small number of tools and can be circumvented (with some effort) by renaming the original tool output in the tool wrapper.

**Comment by jmchilton** [comment]
I think I didn't fix this because I found some tools use the fact these are not quoted to catch wildcards. I think perhaps we should break those tools and just quote this but some people might object. It should be fixed though - this is a problem regardless.

**Comment by natefoo** [comment]
Whatever we do we need to settle on consistent behavior between Galaxy and Pulsar, because in Pulsar the paths are quoted and these tools fail (see galaxyproject/pulsar#239). Not quoting is possibly conceivably somehow unsafe. That said, quoting will break a lot of tools that currently work in Galaxy.

**Comment by jmchilton** [comment]
Those tools are out of spec, it isn't Pulsar's fault that Galaxy is passing them 😅. I guess I'll open a PR to ensure they fail if they have a newer profile version in Galaxy. So from_work_dir will be quoted in Galaxy in 21.05?

**Comment by natefoo** [comment]
> I guess I'll open a PR to ensure they fail if they have a newer profile version in Galaxy. So from_work_dir will be quoted in Galaxy in 21.05?

:+1:

**Comment by natefoo** [comment]
The original issue is fixed by #11960 and #12049. There are no changes to the tool profile yet, so `from_work_dir` values containing wildcards/globs continue to be supported in Galaxy (but not Pulsar) with all current tool profiles. I'll create a new issue for changing that.
